using-valtimo/form-flow/create-form-flow-definition.md

```diff
@@ -0,0 +1,209 @@
+
+    ```json
+    {
+        "startStep": "start",
+        "steps": [
+            {
+                "key": "start",
+                "nextStep": "step2"
+            },
+            {
+                "key": "step2",
+                "nextSteps": [
+                    {
+                        "step": "step3"
+                    },
+                    {
+                        "step": "end"
+                    }
+               ]
+            },
+            {
+                "key": "step3",
+                "nextStep": "end"
+            },
+            {
+                "key": "end"
+            }
+        ]
+    }
+    ```
+
+* Configure the step type. Currently, the only step type supported is `form`, which requires a `definition` property to
+be set. This refers to the key of the form.
+
+    ```json
+    {
+        "startStep": "start",
+        "steps": [
+            {
+                "key": "start",
+                "type": {
+                    "name": "form",
+                    "properties":  {
+                        "definition": "start-form"
+                    }
+                },
+                "nextStep": "step2"
+            },
+            {
+                "key": "step2",
+                "type": {
+                    "name": "form",
+                    "properties":  {
+                        "definition": "step2-form"
+                    }
+                },
+                "nextSteps": [
+                    {
+                        "step": "step3"
+                    },
+                    {
+                        "step": "end"
+                    }
+               ]
+            },
+            {
+                "key": "step3",
+                "type": {
+                    "name": "form",
+                    "properties":  {
+                        "definition": "step3-form"
+                    }
+                },
+                "nextStep": "end"
+            },
+            {
+                "key": "end",
+                "type": {
+                    "name": "form",
+                    "properties":  {
+                        "definition": "end-form"
+                    }
+                }
+            }
+        ]
+    }
+    ```
+
+* Add triggers to steps where necessary (e.g. to store data externally)
+
+
+
+## Expressions
+
+Form flow supports Spring Expression Language (SpEL) expressions in order to allow for more
+complex actions when a step is opened or completed. Expressions can be recognized by the surrounding
+`${ }` characters. The following additional properties are supported for steps:
+
+* `onOpen`. Triggers the expressions when the step is opened, e.g. to retrieve external data.
+
+* `onComplete`. Triggers the expressions when the step is complete, e.g. to store the results in a 
+document.
+
+Each of these properties supports more than one expression, e.g. when a step is opened, external data
+from more than one source is retrieved.
```

Comment:
I would also explain that multiple expressions on a single onOpen of onComplete will be triggered in the order they are added to the array